tests/app/test_settings.py

```diff
@@ -24,5 +25,11 @@ def test_missing_get_application_version_from_file(self):
     def test_none_filename_does_not_attempt_to_load_file(self):
         self.assertEqual(None, settings.read_file(None))
 
+    def test_invalid_key_raises_exception(self):
+        with self.assertRaises(MissingSettingException) as exception:
+            settings.get_env_or_fail("InvalidKey")
```

Comment:
Minor point, but it would be clearer if this was something like `MISSING_ENVIRONMENT_VARIABLE` rather than `InvalidKey`.